birdhouse/components/stac/config/magpie/stac_hooks.py

```diff
@@ -81,64 +64,84 @@ def create_item_resource(response):
-        for childs in children.values():
-            if childs["node"].resource_name == "stac":
-                # find "stac/stac/collections"
-                for stac_child in childs["children"].values():
-                    if stac_child["node"].resource_name == "collections":
-                        # find the id of the resource matching stac/stac/collections/<collection_id>
-                        for collection_child in stac_child["children"].values():
-                            if collection_child["node"].resource_name == collection_id:
-                                collection_res_id = collection_child["node"].resource_id
-                                # find the id of the resource matching stac/stac/collections/<collection_id>/items
-                                for item_child in collection_child["children"].values():
-                                    if item_child["node"].resource_name == "items":
-                                        item_res_id = item_child["node"].resource_id
-                                        item_res_create = False
-                                        break
-
-        # create resource /stac/stac/collections/<collection_id>/items if does not already exist
-        if item_res_create:
-            item_res = ru.create_resource("items", None, Route.resource_type_name, collection_res_id, db_session=session)
-            item_res_id = item_res.json["resource"]["resource_id"]
-
-        # create resource stac/stac/collection/<collection_id>/items/<item_id>
-        # In try catch since Magpie resource are persistent but stac-db could be wiped
-        try:
-            item_res = ru.create_resource(item_id, display_name, Route.resource_type_name, item_res_id, db_session=session)
-        except Exception as exc:
-            LOGGER.warning("Failed creation of the item %s %s", display_name, str(exc), exc_info=exc)
-        session.commit()   
-   
+        # Create the resource tree
+        create_resource_tree(f"stac/collections/{collection_id}/items/{item_id}", children, 0, service.resource_id, session, display_name)
+        session.commit()
+
     except Exception as exc:
         LOGGER.error("Unexpected error while creating the item %s %s", display_name, str(exc), exc_info=exc)
-    
+        session.rollback()
+        return response
+
     return response
 
 def extract_display_name(links):
-    # type: (JSON array) -> string
+    # type: (List[Dict[str, str]]) -> str
     """
     Extract THREDD path from a STAC links
     """
     display_name = None
     for link in links:
-        if link['rel'] == 'source':
+        if link["rel"] == "source":
             # Example of title `thredds:birdhouse/CMIP6` -> `birdhouse/CMIP6`
-            display_name = link['title'].split(':')[1]
+            display_name = link["title"].split(":")[1]
             break
+    if display_name is None:
+        raise Exception("The display name was not extracted properly")
```

Comment:
Better to check onky `if display_name`, this way we reject empty value as well. 
Can raise `ValueError` which is more appropriate than the generic `Exception`.

Reply:
Did you mean ` if not display_name` raise or `if display_name` return and then raise the error ?

Reply:
Yes. My bad. `if not display_name`.